torch/nn/utils/rnn.py

```diff
@@ -1,24 +1,21 @@
-from collections import namedtuple
 import warnings
+from typing import Iterable, List, NamedTuple, Tuple, Union
 
 import torch
 from torch import Tensor
 from ... import _VF
 from ..._jit_internal import Optional
 
-from typing import List, Tuple, Union, Iterable
-
 
 __all__ = ['PackedSequence', 'invert_permutation', 'pack_padded_sequence', 'pad_packed_sequence', 'pad_sequence',
            'unpad_sequence', 'pack_sequence', 'unpack_sequence']
 
-PackedSequence_ = namedtuple('PackedSequence_',
-                             ['data', 'batch_sizes', 'sorted_indices', 'unsorted_indices'])
 
-# type annotation for PackedSequence_ to make it compatible with TorchScript
-PackedSequence_.__annotations__ = {'data': torch.Tensor, 'batch_sizes': torch.Tensor,
-                                   'sorted_indices': Optional[torch.Tensor],
-                                   'unsorted_indices': Optional[torch.Tensor]}
+class PackedSequence_(NamedTuple):
```

Comment:
@davidberard98 Is there anything we need to verify here to ensure TorchScript doesn't break? (or is CI sufficient?)

Reply:
Inheriting from the typed `NamedTuple` is doing exact same thing as the old approach:

```python
In [1]: from typing import *

In [2]: import torch

In [3]: class PackedSequence_(NamedTuple):
   ...:     data: torch.Tensor
   ...:     batch_sizes: torch.Tensor
   ...:     sorted_indices: Optional[torch.Tensor]
   ...:     unsorted_indices: Optional[torch.Tensor]
   ...:     

In [4]: PackedSequence_.__annotations__
Out[4]: {'data': <class 'torch.Tensor'>, 'batch_sizes': <class 'torch.Tensor'>, 'sorted_indices': typing.Optional[torch.Tensor], 'unsorted_indices': typing.Optional[torch.Tensor]}
```

Reply:
looks fine for me re: torchscript (at least the changes in this file with NamedTuple)